sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingQueryListenerBus.scala

```diff
@@ -110,7 +116,15 @@ class StreamingQueryListenerBus(sparkListenerBus: LiveListenerBus)
       listener: StreamingQueryListener,
       event: StreamingQueryListener.Event): Unit = {
     def shouldReport(runId: UUID): Boolean = {
-      activeQueryRunIds.synchronized { activeQueryRunIds.contains(runId) }
+      // When loaded by Spark History Server, we should process all event coming from replay
+      // listener bus.
+      if (sparkListenerBus.isEmpty) {
```

Comment:
In live ui, streaming post event into `StreamingQueryListenerBus`. `StreamingQueryListenerBus` will manage the query status. But in history ui, `StreamingQueryListenerBus` subscribes to the `ReplayListenerBus`, it should process all events.